proposals/nnnn-distributed-actors.md

```diff
@@ -0,0 +1,1620 @@
+
+```swift
+extension Capybara {
+  distributed func meet(friend: ActorIdentity) { ... }
+}
+
+func example3(transport: ActorTransport, friend name: ActorIdentity) async throws {
+  // make local instance
+  let a = Capybara(using: transport)
+
+  // resolve some other instance
+  let b = try Capybara.resolve(name, using: transport)
+  
+  // identities are Codable
+  try await a.meet(b.id)
+
+  // resolve a local instance from an ID
+  let aliasForA = try Capybara.resolve(a.id, using: transport)
+}
+```
+
+The examples so far have assumed that the actor's identity has already been obtained from *somewhere*. But, for a truly remote instance located in a different process, how does the current process obtain that actor's identity, in order to resolve it? 
+
+This task of obtaining or exchanging actor identities is left to specific transport libraries, because it falls under the general problem of service discovery, whose solutions differ based on usage environments. In server environments, it is typical to have a single, fixed location where identity lookups may be performed. For example, those locations can be key-value databases, DNS lookups, or any other more specialized technique. In Swift's existing server ecosystem, the [swift-service-discovery](https://github.com/apple/swift-service-discovery) library can be used to exchange actor identities with a number of service discovery implementations. In the future, it is expected that additional actor-specialized libraries will become available.
+
+### Distributed Actor Example
+
+Now that we know distributed actor basics, we can rewrite the RPC example from earlier using distributed actors. We do so by changing `actor Counter`  to `distributed actor Counter`, and marking the `increment` and `get` methods as `distributed` too. This is all we need to do to prepare this actor for distributed calls.
+
+```swift
+import _Distributed
+
+distributed actor Counter { 
+  var count: Int = 0
+  
+  distributed func increment() {
+    count += 1
+  }
+  
+  distributed func get() -> Int { 
+    return count
+  }
+}
+```
+
+For the RPC example, some process must first locally create an instance of the distributed actor `Counter`. This is done by calling`Counter.init(transport:)` and passing the specific transport we want it to accessed through:
+
+```swift
+import FishyTransport 
+
+let transport = FishyTransport(...)
+let counter = Counter(transport: transport)
+
+/* Optional: register counter.id in some service discovery mechanism */
+// discovery.announce("Counters", counter.id)
+```
+
+
+Once we have either obtained the specific identity on a remote node, we can resolve it to get a reference to the remote actor, like this:
+
+```swift
+let id: AnyActorIdentity = try FishyTransport(host: "example.com", port: 1337, logLevel: nil)
```

Comment:
This example seems to mix Transport and AnyActorIdentity in a non-obvious way. Maybe that distinction could be made clearer, e.g. by using `FishyTransport.Identity(host: …)`

Reply:
Oh thanks, somethign went wrong here, this should be just "imagine we got an identity"

```suggestion
let id: AnyActorIdentity = ...
```